api/src/pcapi/scripts/business_unit/business_unit_builder.py

```diff
@@ -0,0 +1,632 @@
+
+            main_venue = None
+            if not siret_pre_bu_status:
+                main_venue = Venue.query.filter(
+                    Venue.managingOffererId == self.offerer.id,
+                    Venue.siret == main_siret,
+                ).one_or_none()
+                if main_venue and not main_venue.businessUnitId:
+                    main_venue_assignable = True
+
+            if main_venue_assignable:
+                pre_bu.siret = main_siret
+                pre_bu.venues.append(main_venue)
+                pre_bu.fill_data_from_siret(main_siret)
+                pre_bu.status = PreBusinessUnitStatus.READY_API_SIRET_MATCH_MANAGED_VENUE
+            else:
+                pre_bu.status = PreBusinessUnitStatus.END_API_SIRET_MATCH_MANAGED_VENUE_NOT_ASSIGNABLE
+
+    def process_draft_api_siret_not_match_managed_venue(self):
+        '''
+        The main SIRET doesn't exist on offerer's managed venue
+        Here we try to match main venue address within PreBusinessUnit venues
+        '''
+        siren_api_data = self.get_main_venue_siren_api_data()
+        main_siret = siren_api_data["siret"]
+        main_address = siren_api_data["address"]
+        main_postal_code = siren_api_data["postalCode"]
+        main_city = siren_api_data["city"]
+
+        business_unit_list = self.get_for_statuses([PreBusinessUnitStatus.DRAFT_API_SIRET_NOT_MATCH_MANAGED_VENUE])
+        for pre_bu in business_unit_list:
+            venue_match_address = []
+            for venue in pre_bu.venues:
+                if venue.siret:
+                    continue
+
+                if all(check for check in [
+                    main_address == venue.address,
+                    main_postal_code == venue.postalCode,
+                    main_city == venue.city,
+                ]):
+                    venue_match_address.append(venue)
+            if len(venue_match_address) == 1:
+                pre_bu.siret = main_siret
+                main_venue = venue_match_address[0]
+                pre_bu.main_venue_id = main_venue.id
+                pre_bu.status = PreBusinessUnitStatus.PENDING_API_ADDRESS_MATCH
+            else:
+                pre_bu.status = PreBusinessUnitStatus.DRAFT_API_ADDRESS_NOT_MATCH
+
+    def process_pending_api_address_match(self):
+        '''
+        Main venue have been found in PreBusinessUnit venue by matching the address.
+        Here we assign main the main SIRET to this venue
+        '''
+        business_unit_list = self.get_for_statuses([PreBusinessUnitStatus.PENDING_API_ADDRESS_MATCH])
+        for idx, pre_bu in enumerate(business_unit_list):
+            # we cannot create offere main venue a second time
+            if idx > 0:
+                pre_bu.status = PreBusinessUnitStatus.END_API_ADDRESS_MATCH_VENUE_NOT_ASSIGNABLE
+            else:
+                main_venue = Venue.query.get(pre_bu.main_venue_id)
+                main_venue.siret = pre_bu.siret
```

Comment:
J'ajouterais bien une vérification que le `siret` est vide, ici (pour éviter d'écraser un SIRET en cas de bug dans le script) :

```suggestion
                if main_venue.siret:
                    raise ValueError(f"Trying to overwrite siret of venue {main_venue.id} from {main_venue.siret} to {pre_bu.siret}")
                main_venue.siret = pre_bu.siret
```

Reply:
Hum, 
ok mais ligne 394, avant de passer en status `PENDING_API_ADDRESS_MATCH` on à déjà un `if venue.siret: continue`.